src/middleware/identity/impl.js

```diff
@@ -44,6 +45,7 @@ module.exports = function (req, res, next) {
                 username: req.identity.service_account.username,
                 is_internal: false
             };
+            req.type = "ServiceAccount"
```

Comment:
This looks good but I'm wondering why we can't just check the value of `req.identity.type` instead of adding the `req.type`? 🤔 I'm okay with `req.type` but just curious if there's a reason for using it instead of `req.identity.type`

Reply:
I believe @rexwhite had suggested something like this in case the syntax in the request headers that we receive ever changed so we would only have to update that in one place instead of multiple.

I was originally going to use something like `req.is_service_account = req.identity.type == "ServiceAccount" ? true : false;` in `src/middleware/identity/impl.js` and check that respectfully in other places instead of assigning `req.type` and checking that. We can go with that if you think it's cleaner.

Reply:
I kind of like the idea of checking the actual type in case there's other types we want to check for in the future besides `User` or `ServiceAccount` 

I was thinking that we could just check if `req.identity.type == "ServiceAccount"` in `controller.read.js` but it's true that if `req.identity.type` changes in the future it might be easier to just have to update it in this file and `req.type` would always stay the same. So I think this is probably best!